### PR TITLE
Fix Array/Map DataType result mismatch issue

### PR DESCRIPTION
**Pull request by zhixingheyi-tian**
## What changes were proposed in this pull request?

Just fix issue #877 , for detail, can refer to #877

## How was this patch tested?

(Please explain how this patch was tested. E.g. unit tests, integration tests, manual tests)

(If this patch involves UI changes, please attach a screenshot; otherwise, remove this)

### CODE REVIEW

**Comment by zhixingheyi-tian** [comment]
cc @zzcclp  @rui-mo

**Review by zzcclp — APPROVED** [review]
LGTM

File: gluten-data/src/main/scala/io/glutenproject/execution/GlutenRowToArrowColumnarExec.scala [hunk 1]
```diff
@@ -60,12 +60,12 @@ object RowToColumnConverter {
       case StringType => StringConverter
       case BinaryType => BinaryConverter
       case CalendarIntervalType => CalendarConverter
-      case at: ArrayType => new ArrayConverter(getConverterForType(at.elementType, nullable))
+      case at: ArrayType => new ArrayConverter(getConverterForType(at.elementType, at.containsNull))
       case st: StructType => new StructConverter(st.fields.map(
         (f) => getConverterForType(f.dataType, f.nullable)))
       case dt: DecimalType => new DecimalConverter(dt)
       case mt: MapType => new MapConverter(getConverterForType(mt.keyType, nullable),
```
Comment (jinchengchenghh) [inline gluten-data/src/main/scala/io/glutenproject/execution/GlutenRowToArrowColumnarExec.scala:67]: The MapType key nullable check is right?

Comment (zhixingheyi-tian) [reply gluten-data/src/main/scala/io/glutenproject/execution/GlutenRowToArrowColumnarExec.scala]: Key can't be null, just set false

**Review by jinchengchenghh — APPROVED** [review]

**Pull request merged**